internal/storage/storage.go

```diff
@@ -46,6 +54,18 @@ type QueryParams struct {
 	Order     Order // not exposed to the user yet
 }
 
+func (q *QueryParams) Validate() error {
+	if q.Limit == 0 {
+		q.Limit = DefaultListLimit
+	}
+
+	if q.Limit > MaxListLimit {
+		return fmt.Errorf("maximum page size limit (%d) exceeded: %d", MaxListLimit, q.Limit)
```

Comment:
Yeah, only if we introduced `Validate()` call in the existing packages. Which I haven't done in this PR.
If we're worried about inadvertently introducing that later, I could move this into the auth package?

Reply:
r.e. should we just snap to the max limit? Good question.

At-first, my feelings were that is implicit behaviour that might be surprising to an end user.
I request 1000 and get back 100.

However, I can see how this might be more jarring; I request 1000, get back a 4XX error (which I think I haven't actually done the work to make it a validation error—I would do that if we keep down this path).
I have to read the error and adjust my code to match limits.

I think if pagination was the way it used to be, I would be less inclined to snap the page to max.
As this could lead to mistakes when implementing pagination (end-user _must_ rely on response length to set next offset; they might mistakenly use `old_offset + limit` when requesting next page).
Given pagination is driven by page token, there is less meaning to the number of results in the response.
If you want more or all, you just keep following the page tokens until they end.
You can request 1000, but you will only get the max page size in any given response. Not an error.

TL;DR I will do as suggested and snap limit to max 👍

Reply:
See commit 5f94153e where I renamed `Validate()` to `Normalize()` and drop the error for this change 👍